core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts

```diff
@@ -133,6 +134,70 @@ angular.module('oppia').directive('topicEditorTab', [
             $scope.topicUrlFragmentExists = false;
             $scope.hostname = WindowRef.nativeWindow.location.hostname;
 
+            $scope.getEligibleSkillSummariesForDiagnosticTest = function() {
+              let availableSkillSummaries = (
+                $scope.topic.getAvailableSkillSummariesForDiagnosticTest());
+
+              return availableSkillSummaries.filter(skillSummary => {
+                return (
+                  $scope.skillQuestionCountDict[skillSummary.getId()] >=
+                  MINIMUM_QUESTION_COUNT_FOR_A_DIAGNOSTIC_TEST_SKILL
+                );
+              });
+            };
+            $scope.availableSkillSummariesForDiagnosticTest = (
+              $scope.getEligibleSkillSummariesForDiagnosticTest());
+            $scope.selectedSkillSummariesForDiagnosticTest = (
+              $scope.topic.getSkillSummariesForDiagnosticTest());
+            $scope.diagnosticTestSkillsDropdownIsShown = false;
+            $scope.selectedSkillForDiagnosticTest = null;
+
+            $scope.presentDiagnosticTestSkillDropdown = function() {
+              $scope.diagnosticTestSkillsDropdownIsShown = true;
+            };
+
+            $scope.removeDiagnosticTestSkillDropdown = function() {
+              $scope.diagnosticTestSkillsDropdownIsShown = false;
+            };
+
+            $scope.addSkillForDiagnosticTest = async function() {
+              let skillToAdd = $scope.selectedSkillForDiagnosticTest;
+              $scope.selectedSkillForDiagnosticTest = null;
+
+              $scope.selectedSkillSummariesForDiagnosticTest.push(skillToAdd);
+              let j = -1;
```

Comment:
Could you use a more descriptive name than "j"? Ditto below.

I also wonder whether find() would be useful here -- you could return the index and then do the splice after the find(), perhaps. Might be easier to read.

Reply:
Updated with the `find` method.